### DISCUSSION TITLE
Question about Story Progression [Major Spoilers]

### DISCUSSION THREAD

**Discussion by Quotatron3000** [endless-sky/endless-sky#10227] [Q&A]
Spoiler warning for anyone playing Endless Sky and don't want future plots ruined, this contains major spoilers.

_Just to be clear, I'm intentionally avoiding any sort of online strategy guides because I'm trying to learn the game based on the game itself as a de facto "beta test" on user experience._

I'm currently in Coalition space, and I've gone down the Lunarium pathroute. The story path progression has, to my understanding, told me that after I distribute ship parts/weapons to various planets for a while, I will eventually get contacted again. It has not given me a quota of how many of these grind missions I need to complete before the story will progress.

I've been doing this for several real days worth of effort now and the story does not seem to have progressed, and I haven't been contacted by the Lunarium. Either I've missed or screwed up some sort of vital story trigger, or this part of the game is still under development.

I've also seen the Hai progression seemingly stall. It was left that the Republic were still in tension about the discovery of the Hai, and I completed a number of diplomatic missions before it suggested that I should 'explore Hai space' and that they would 'get back in touch', but this also hasn't seemingly happened. I don't know if this is separate or related, but the last mission I did involved taking some sort of Hai prophet to Unfettered space. For what it is worth, I had _already explored Hai space_ before any of the jump drive related missions occurred, so I'm wondering if I might have pre-triggered a set trigger event before the game was ready.

So to summarise my questions:

1. Are the Lunarium missions still in development at the 'send ship parts/supplies' stage? If not, what is the trigger for story progression (because it isn't this endless shipping).
2. Are the Hai missions still in development at the end of the diplomacy/Hai discovery crisis? If not, what's the trigger?
3. Are there any means of debugging I can use given these issues have happened quite some time ago?

I don't mind if it's still in development, I just don't want to waste my time searching for an outcome that doesn't exist.

**Comment by Arachi-Lover** [comment]
On the Lunarium missions, the grinding isn't a requirement, if you mean the smuggling jobs from the Job Board. Those are simply unlocked jobs you can do for some money if you'd like. The requirements for the other missions in that chain, as well as other mission chains for the Lunarium, are as follows:

```
Lunarium side

-Intro (source - requirements)-
Smuggling - Kimek space at first, then all over the Coalition - Requires first contact at first, then later requires Deep Sky tech and Flamethrowers to be purchaseable.
Bebliss - Anywhere other than Factory of Eblumab - Requires the Coalition License, and Freya's epilogue.
Evacuation - Kimek space, then Saryd space, then Arach space - Requires just the first contact.
Propaganda - 2 to 4 jumps from Ancient Hope - Requires just the first contact.
Combat Training - Arach space - Requires having completed at least any 1 of the following chains: Smuggling, Evacuation, Propaganda, as well as not having some ongoing FW event in Shaula.

Questions (joining them) - Anywhere other than Remote Blue - Requires positive Quarg rep, and either having completed all 5 chains, having completed 4 + 80 jobs, or having completed 3 + 110 jobs. 
```

If you'd like, the Heliarch side's missions are similarly like this:

```
-Intro (source - requirements)-
Investigation chain - 4 to 10 jumps from Quaru - Requires 65 Coalition jobs and the Coalition License
Recon - One of the Heliarch rings - Requires positive Quarg reputation, requires 55 Coalition jobs, the Coalition License, having had First Contact with the Hai, having visited Lagrange, Alta Hai and Kuwaru Efreti, and having a JD. Later on in the chain it is required to have gotten a Quarg dialog you get by landing on Quarg areas and being questioned by them, and having completed Rim Archaeology (meaning there's a Wardragon flying over Zug).
Expedition - Ring of Wisdom - Requires having completed the Recon chain, and having landed on Ruin
Containment - Kimek space, then Arach space, then the Ring of Power - Requires just the first contact at first, later requires that the Plasma Turret is available for purchase
Drills - 6+ jumps from Belug - Requires CR >= 8104 (CR rank 9).

License (joining them) - Anywhere other than Quaru - Requires either having completed all 5 chains, having completed 4 + 90 jobs, or having completed 3 + 120 jobs.
```

I've also made this diagram a while ago that showcases the same information as those text bits, if it's easier to read:
![image](https://github.com/endless-sky/endless-sky/assets/82293490/fc512334-87c6-4ac4-a33d-2aaad6deb60e)

**Reply by Quotatron3000** [reply]
I must profess some confusion.

For me, the smuggling jobs appeared _after_ I joined the Lunarium, and the in-game story makes it seem like you have to complete a number of smuggling missions before the story progresses.

What I'm expecting to see is either the next stage of their war plans (I dunno, maybe... delivering specific ships? Or acquiring ship blueprints?)... or what I assume will become the war between them and the Heliarchy.

===

Whilst I'm here I forgot to mention - Remnant progression for a Remnant licence isn't clear either. I scanned one of their ships and started a shooting war, but after leaving the system they seemed... okay with it? All I can do is job board trades, and spaceport doesn't offer anything.

**Reply by Arachi-Lover** [reply]
After you join, if you didn't do all 5 mission chains for one side or the other, you can the do said missions for the side you chose, and most have some changes recognizing you've already joined (just different text, basically).

What I posted were the mission sets for the Intros of both campaigns. They are all that's in the game right now, as the next parts aren't even written yet.

**Reply by Quotatron3000** [reply]
I figured it might not have been complete given the Free Worlds Fourth Wall Break, where it mentions other content isn't yet available.

Is there any documentation where I can learn how to contribute? I'd love to help build out the Coalition story, at the very least for the Lunarium. I've got quite a few ideas on what I expected to happen next, but I think it'd be more useful if I helped to implement more of the story.

My rough sketch on progression was:
- House Bebliss contacts you after X delivery missions (between 10 to 20)
- They ask for ship blueprints from across the galaxy (Free Worlds, Republic, Syndicate, and optionally Hai if discovered)
- Optional side mission where you can supply them with examples of Hai or Korath anti-missile systems (contingent on you having seen either) after noting that human missile defences are inferior. The only difference is Lunarium ships will make use of these instead of human anti-missile defences when the war occurs.
- Heliarch spot your activity and make you an offer to become a turncoat (for simplicity, you'd automatically refuse at this juncture, but it would allow the expansion of additional 'traitor' content; alternatively the player could 'accept' and only pretend to be a turncoat - either way, interesting outcomes). If you refuse, Lunarium agents bust you out and tell you to flee to human space as Coalition turns hostile.
- You get contacted by either the Free Worlds or the Republic asking 'what the heck are you doing starting a civil war?'. The Republic doesn't like the idea, the Free Worlds 'understands why you're doing it'.
- Lunarium asks you to act as ambassador and ask for human assistance.
- Free Worlds offers to assist by supplying dreadnought designs and 'old ships they have lying around'. Syndicate smell a money opportunity and are happy to supply a stream of weapons and even possibly ships contingent on Bebliss agreeing to a trade contract.
- The "oynx black" Quarg mentioned earlier becomes a 'Chekov's gun', after it is found to be a mixture of depression and anger. Whilst a number of the Quarg won't give away their own technology, there is nothing stopping them from souping up either human or Lunarium technology. Whilst they won't get involved in bloodshed, they will give the Lunarium special anti-missile jamming technology that works very effectively against Coalition ships' missiles, on the condition Quarg are allowed to return to the ring if they so choose.
- The player is told to build up a fleet of ships as the Coalition ships are very tough. They're advised to install torpedoes.
- Lunarium, now confident, declare war on Heliarch.
- An announcement goes out on the various houses and/or planets pledging allegiance to the sides they choose. Some, like the bankers, choose to remain neutral. The ring worlds turn hostile to the player. The player has the task of convincing the neutral planets to join the Lunarium, all whilst the battle rages on.
- The player is required to attend a massive fleet engagement between Lunarium and Heliarchs to help turn the tide.
- Similar to Free Worlds, planets are either diplomatically won over or conquered, however the player discovers, again, the people giving Lunarium 'accurate information' is the same Oracle run by the Alphas that gave the Syndicate information, exposing a wider plot.
- It turns out the Alphas goal is to weaken the Coalition internally, like they did human space, in preparation for some sort of 'big event'.
- The player experiences an assassination attempt by a Heliarch agent that infiltrated their ship. They have to carefully choose a set of dialogue prompts to determine the outcome.
- As Heliarchs lose, player becomes an ambassador for peace and reconciliation. The player is rhetorically hit over the head over the consequences of their actions, the sheer amount of devastation, lives lost, as they travel from planet to conquered planet.
- The Heliarchs eventually surrender the ringworlds on the condition of bulk amnesty, with Lunarium arguing no amnesty for those involved in war crimes.
- Coalition space opens up, and the Free Worlds establishes a hyperspace lane to Coalition space from Poisonwood (contingent on completing the Pug mission)
- Free Worlds, Republic and Syndicate all send their respective 'ambassadors'. Republic sends a politician, Free Worlds a diplomat, and the Syndicate a contract negotiator. If Hai have been revealed, the Hai also send an ambassador as well.
- Coalition space returns to 'friendly' for the player, besides any neutral or destroyed worlds.
- Player receives a Lunarium licence, allowing them to buy the modified Lunarium ships, which includes modified Coalition ships (namely, the very large warships).

I know it's a lot but I was trying to anticipate story progression for a while.

**Reply by ziproot** [reply]
If you aren’t on the [Discord](https://discord.gg/ZeuASSx), I would recommend joining it. There are people on there who are happy to help you contribute to the game.

**Reply by Arachi-Lover** [reply]
I've got a plan for the Coalition campaigns in mind, and at least a rough sketch of each mission for the next portion of both campaigns, but the Quarg rework/expansion is a necessity before a lot of the remaining bits of the campaign could be tested properly, so it'll probably be a while still before those are ready.

As mentioned you could check on the Discord server for general ideas and discussion, if you'd like to make content of your own.
For the Coalition, there's also the Ideas tab here: https://github.com/orgs/endless-sky/projects/7/views/2 , where some scrapped plans of mine are, if you'd like to check there for some potential inspiration or story hooks.

**Comment by TheGiraffe3** [comment]
For the Hai part, a rework is planned for it, and so some of the missions are disabled. So basically you’re at the end of Hai Reveal (HR) right now. There is a plugin someone made that enables it to continue though if you want, [here](https://github.com/endless-sky/endless-sky/discussions/9161#discussioncomment-7041691) (though it was made in 0.10.2).  

About the Remnant, have you checked the spaceport on all three major planets of theirs?

**Comment by xX-Dillinger-Xx** [comment]
It seems like the underlying issue is that OP reached the "current" end of both story lines. The problem is the game does not inform the player they have reached the end and the players keeps pounding away, hoping to continue at some point. 

I have a suggestion to help end this type of confusion. At the end of any unfinished story lines there should be a message that states something like `Hello player, it appears you have reached the end of this currently unfinished storyline. There will be more content in the future.`
